cuML/src/glm/qn/simple_mat.h

```diff
@@ -162,41 +163,39 @@ template <typename T> struct SimpleMat {
                                 len, f, stream);
   }
 
-  inline void fill(const T val, cudaStream_t stream=0) {
+  inline void fill(const T val, cudaStream_t stream) {
     // TODO this reads data unnecessary, though it's mostly used for testing
     auto f = [val] __device__(const T x) { return val; };
     MLCommon::LinAlg::unaryOp(data, data, len, f, stream);
   }
 
-  inline void operator=(const SimpleMat<T> &other) {
+  inline void copy(const SimpleMat<T> &other, cudaStream_t stream) {
+    ASSERT((ord == other.ord) && (m == other.m) && (n == other.n),
+           "SimpleMat::copy: matrices not compatible");
 
-    ASSERT(ord == other.ord, "SimpleMat::operator=: Storage orders must match");
-
-    CUDA_CHECK(cudaMemcpy(data, other.data, len * sizeof(T),
-                          cudaMemcpyDeviceToDevice));
+    CUDA_CHECK(cudaMemcpyAsync(data, other.data, len * sizeof(T),
+                               cudaMemcpyDeviceToDevice, stream));
+    CUDA_CHECK(cudaStreamSynchronize(stream));
```

Comment:
`copy` takes a stream as an argument. From a callersperspetive this expresses that it is an asynchronous operation. Please remove the `cudaStreamSynchronize`. This is something the caller want's to control. If you want to be more explicit about this add an async suffix to copy.

Reply:
Good point!